design/thin-edge-core.md

```diff
@@ -0,0 +1,50 @@
+# Thin-edge Core Vision
+
+A typical IoT application has to address:
+
+* __sophisticated business-cases__, with applications involving several vendors, actors and users;
+* __very diverse integration needs__ over operating systems, protocols, data sources, processing tools, actuators, cloud end-points;
+* __constraint environments__ with low-resources devices, large-scale deployments, high exposure to security risks,
+  prohibitive manual operations, long-term supports;
+* __safety related requirements__ with devices operating machines in hazardous contexts.
```

Comment:
I like to add here:

* __access to existing untouchable machinery/devices__, that cannot be changed/extended cause they are closed entirely HW/SW systems; or shall not be changed/extended cause they are well-proven and any risk coming from changes must be avoided;

Reply:
This is indeed a constraint. I updated the text to briefly convey the point with legacy systems and restricted upgrades.